### PR TITLE
Use extra-* in nix.extraOptions to append to existing entries

### LINKED ISSUE #91: nix config is overriding
https://github.com/divnix/digga/blob/f25928ed0bf05ab20af5a8231d264ae01b42ab34/modules/nix-config.nix#L23-L26

does not behave well with the current implementation of this option, since later values seem to override previous values: https://github.com/NixOS/nixpkgs/blob/b2f87e0043aaf3f0f05cc983bd6aa80a616b8352/nixos/modules/services/misc/nix-daemon.nix#L59

### PR DESCRIPTION
**Pull request by teutat3s**
Possibly fixes #91 

This PR tries to fix issues that appeared when testing the `develop` branch, where devshell started to build gcc instead of using the binary caches. Upon inspection of `nix.conf` it looks like there are duplicate entries probably overriding each other:

```
# WARNING: this file is generated from the nix.* options in
# your NixOS configuration, typically
# /etc/nixos/configuration.nix.  Do not edit it!
build-users-group = nixbld
max-jobs = auto
cores = 0
sandbox = true
extra-sandbox-paths = 
substituters = https://nrdxp.cachix.org https://nix-community.cachix.org https://cache.nixos.org/ https://cache.nixos.org/
trusted-substituters = 
trusted-public-keys = nrdxp.cachix.org-1:Fc5PSqY2Jm1TrWfm88l6cvGWwz3s93c6IOifQWnhNW4= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
auto-optimise-store = true
require-sigs = true
trusted-users = root @wheel
allowed-users = @wheel
builders =

system-features = nixos-test benchmark big-parallel kvm
sandbox-fallback = false

min-free = 536870912
keep-outputs = true
keep-derivations = true
fallback = true

experimental-features = flakes nix-command ca-references
substituters = https://nrdxp.cachix.org https://nix-community.cachix.org
trusted-public-keys = nrdxp.cachix.org-1:Fc5PSqY2Jm1TrWfm88l6cvGWwz3s93c6IOifQWnhNW4= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=

extra-experimental-features = nix-command ca-references flakes

```

This PR hopes to workaround the missing merge semantics in `nix.extraOptions`.

### CODE REVIEW

File: modules/nix-config.nix [hunk 1]
```diff
@@ -20,9 +20,10 @@ in
   # package and option is from fup
   nix.generateRegistryFromInputs = lib.mkDefault true;
 
+  # missing merge semantics in this option force us to use extra-* for now
   nix.extraOptions = ''
-    experimental-features = ${lib.concatStringsSep " " experimental-features }
-    substituters = ${lib.concatStringsSep " " substituters }
-    trusted-public-keys = ${lib.concatStringsSep " " trusted-public-keys }
+    extra-experimental-features = ${lib.concatStringsSep " " experimental-features }
```
Comment (gytis-ivaskevicius) [inline modules/nix-config.nix:25]: Likely to conflict with fup. Maybe we should create some sort of API for this?

**Review by gytis-ivaskevicius — APPROVED** [review]

**Comment by teutat3s** [comment]
When this change testing locally with 
```
sudo nixos-rebuild switch --flake ".#dumpyourvms" --override-input digga github:pub-solar/digga/ad55b3ade6d20a14ca390a2f141c3e63bddda3ee
```

the following `nix.conf` got spit out:
```
# WARNING: this file is generated from the nix.* options in
# your NixOS configuration, typically
# /etc/nixos/configuration.nix.  Do not edit it!
build-users-group = nixbld
max-jobs = auto
cores = 0
sandbox = true
extra-sandbox-paths = 
substituters = https://nrdxp.cachix.org https://nix-community.cachix.org https://cache.nixos.org/ https://cache.nixos.org/
trusted-substituters = 
trusted-public-keys = nrdxp.cachix.org-1:Fc5PSqY2Jm1TrWfm88l6cvGWwz3s93c6IOifQWnhNW4= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
auto-optimise-store = true
require-sigs = true
trusted-users = root @wheel
allowed-users = @wheel
builders =

system-features = nixos-test benchmark big-parallel kvm
sandbox-fallback = false

min-free = 536870912
keep-outputs = true
keep-derivations = true
fallback = true

extra-experimental-features = flakes nix-command ca-references
extra-substituters = https://nrdxp.cachix.org https://nix-community.cachix.org
extra-trusted-public-keys = nrdxp.cachix.org-1:Fc5PSqY2Jm1TrWfm88l6cvGWwz3s93c6IOifQWnhNW4= nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=

extra-experimental-features = nix-command ca-references flakes
```

**Comment by teutat3s** [comment]
I'm not sure if this is a "clean" fix, but it works in my local tests.

File: modules/nix-config.nix (hunk 1 above)
Comment (blaggacao) [reply modules/nix-config.nix:25]: That is a good idea! Though I don't think it conflicts with current fup develop, does it?

**Comment by blaggacao** [comment]
bors try

**Review by blaggacao — APPROVED** [review]
LGTM as a temporary work around.

The underlying issue is that later config file values override previous ones.

The root cause of this issue is a non-composable upstream `nix.extraOptions` option.

File: modules/nix-config.nix (hunk 1 above)
Comment (gytis-ivaskevicius) [reply modules/nix-config.nix:25]: Only one `extra-experimental-features` can be valid at the time, thus depending on merge ordering configuration will use option either from fup or devos

**Pull request merged**